crates/interpreter/src/side_table.rs

```diff
@@ -86,6 +88,32 @@ pub struct MetadataEntry {
     pub branch_table: Vec<BranchTableEntry>,
 }
 
+pub fn serialize(side_table: &[MetadataEntry]) -> Result<Vec<u8>, Error> {
+    let mut res = vec![];
+    res.extend_from_slice(&(side_table.len() as u16).to_le_bytes());
+    let mut index = 0;
+    res.extend_from_slice(&(index as u16).to_le_bytes());
+    for entry in side_table {
+        index =
+            try_from::<u16>("index of MetadataEntry", index + 10 + 6 * entry.branch_table.len())?
+                as usize;
+        res.extend_from_slice(&(index as u16).to_le_bytes());
```

Comment:
We should actually have `index: u16`. The casts are at the wrong places. It should be:

```rust
    let mut index = 0u16;
    res.extend_from_slice(&index.to_le_bytes());
    for entry in side_table {
        index = try_from::<u16>(
            "index of MetadataEntry",
            index as usize + 10 + 6 * entry.branch_table.len(),
        )?;
        res.extend_from_slice(&index.to_le_bytes());
    }
```